src/object.c

```diff
@@ -998,8 +997,6 @@ size_t objectComputeSize(robj *key, robj *o, size_t sample_size, int dbid) {
                 samples++;
             } while ((node = node->next) && samples < sample_size);
             asize += (double)elesize/samples*ql->len;
-        } else if (o->encoding == OBJ_ENCODING_ZIPLIST) {
-            asize = sizeof(*o)+zmalloc_size(o->ptr);
```

Comment:
that's actually dead code since 3.2, and objectComputeSize was only introduced in 4.0.
i guess i initially developed #3223 based on 2.8 or 3.0.